libzkbob-rs/src/merkle.rs

```diff
@@ -439,17 +385,37 @@ impl<D: KeyValueDB, P: PoolParams> MerkleTree<D, P> {
 
     // Get root at specified index (without rollback)
     // This method can be used for tree integrity check
-    // index is a position within Merkle tree level 0
-    // and should point to the first transaction leaf
-    //  (to be a multiple of constants::OUT + 1)
+    // `index` is a position within Merkle tree level 0
+    // it should be <= next_index
     pub fn get_root_at(&self, index: u64) -> Option<Hash<P::Fr>> {
-        if index & constants::OUT as u64 != 0 ||
-            index > self.next_index()
-        {
+        if index > self.next_index() {
             return None;
         }
 
-        self.get_with_next_index_recursive(constants::HEIGHT as u32, 0, index)
+        if index == 0 {
```

Comment:
I guess this code:
```
if index == 0 {
   return Some(self.default_hashes[constants::HEIGHT]);
}
let non_zero_sibling_height = index.trailing_zeros() as usize;
```
can be replaced with something like:
```
let non_zero_sibling_height = (index.trailing_zeros() as usize).min(constants::HEIGHT);
```
But it is just a suggestion :)

Reply:
Agree, looks a little simpler this way